src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -4245,6 +4245,7 @@ void Assembler::vpermq(XMMRegister dst, XMMRegister nds, XMMRegister src, int ve
 
 void Assembler::vpermb(XMMRegister dst, XMMRegister nds, XMMRegister src, int vector_len) {
   assert(VM_Version::supports_avx512_vbmi(), "");
+  InstructionMark im(this);
```

Comment:
May be add short comment why you need `InstructionMark` in these instructions but not in others.